pkg/client/leaderelection/leaderelection.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package leaderelection implements leader election of a set of endpoints.
+// It uses an annotation in the endpoints object to store the record of the
+// election state.
+package leaderelection
```

Comment:
Is the wikipedia [summary](https://en.wikipedia.org/wiki/Leader_election) and [definition](https://en.wikipedia.org/wiki/Leader_election#Definition) wrong? 

I think there is value in an api that has higher level semantics than lock/lease nor do I want this to be used or become a generic lock lease/library on kubernetes objects. I still hold that the internal lease is an implementation detail.